### ISSUE TITLE
manual_range_contains misses non-literals

### ISSUE THREAD

**Issue by Niklas-Eckhoff** [rust-lang/rust-clippy#7675]
Lint name: [manual_range_contains](https://rust-lang.github.io/rust-clippy/master/index.html#manual_range_contains)

I tried this code:

```rust
fn main() {
    let x = 4.0;
    let y = 0.0;
    let z = 5.0;
    if y <= x && x <= z {
        println!("Hello, world!");
    }
}
```

I expected to see this happen: *Warning: help: use: `(y..=z).contains(&x)`*

Instead, this happened: *No warning.*

### Meta

**Rust version (`rustc -Vv`):**

```
rustc 1.55.0 (c8dfcfe04 2021-09-06)
binary: rustc
commit-hash: c8dfcfe046a7680554bf4eb612bad840e7631c4b
commit-date: 2021-09-06
host: x86_64-unknown-linux-gnu
release: 1.55.0
LLVM version: 12.0.1
```

**Comment by xFrednet** [comment]
Thank you for the bug report. From a quick test and a look at the [code](https://github.com/rust-lang/rust-clippy/blob/59cd77710d73ba65d56f0c5c167ef503174a4826/clippy_lints/src/ranges.rs#L291-L311), it seems like the current implementation only works with constant values.  Having it work with local values as well sounds reasonable IMO. :upside_down_face:
